### PR TITLE
docs(lambda-destination): add onFailure destination example and be explicit about the type of SQS queue that can be used

### LINKED ISSUE #13793: (aws-lambda-destinations): update docs to warn users that Fifo queues aren't supported as destinations
During creation of a Lambda defined in a Python stack, a status code 400 `InvalidParameterValueException` is thrown. It fails to create the SQS `on_failure` `DestinationConfig` and the stack creation fails with a rollback.

### Reproduction Steps
Deploying the following stack triggers the failure:

```python
from aws_cdk import (
    core,
    aws_lambda as lambda_,
    aws_lambda_destinations as lambda_dest,
    aws_sqs as sqs,
)
from aws_cdk.aws_lambda_python import PythonFunction


class StackHello(core.Stack):

    def __init__(self, scope: core.Construct, construct_id: str,
                 **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self._create_hello_lambda()

    def _create_hello_lambda(self) -> None:
        q = sqs.Queue(
            scope=self,
            id='hello-fail.fifo',
            queue_name='hello-fail.fifo',
            encryption=sqs.QueueEncryption.KMS_MANAGED,
            fifo=True,
            content_based_deduplication=True,
            visibility_timeout=core.Duration.seconds(90)
        )
        
        fn = PythonFunction(
            scope=self,
            id='hello_function',
            handler='main',
            entry='hello',
            index='hello.py',
            runtime=lambda_.Runtime.PYTHON_3_7,
            on_failure=lambda_dest.SqsDestination(q)
        )
        q.grant_send_messages(fn)

```

### What did you expect to happen?
Trying to deploy a Lambda function through the CDK that sends any message that failed to process to a dead letter queue.

### What actually happened?
The deploy fails with the exception below:
```
8:32:00 AM | CREATE_FAILED        | AWS::Lambda::EventInvokeConfig | hello_function/EventInvokeConfig
The provided destination config DestinationConfig(onSuccess=OnSuccess(destination=), onFailure=OnFailure(destination=arn:aws:sqs:us-east-1:<redacted>:hello-fail.fifo)) is invalid. (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueEx
ception; Request ID: bf0267ad-14f8-4d6d-b788-b547e390f8f0)

```

### Environment

  - **CDK CLI Version  :** 1.94.1 (build 60d8f91)
  - **Framework Version:** 1.94.1
  - **Node.js Version:** v15.5.1
  - **OS               :** macOS Catalina (10.15.7)
  - **Language (Version):** Python 3.7.8

### Other
None.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by RichardWatkins1**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

closes #13793

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-destinations/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Destinations can be added by specifying the `onFailure` or `onSuccess` props whe
 The following destinations are supported
 
 * Lambda function
-* SQS queue
+* Standard SQS queue
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda-destinations/README.md:20]: FIFO queues are supported for successful invocations, right?

File: packages/@aws-cdk/aws-lambda-destinations/README.md [hunk 2]
```diff
@@ -38,6 +38,23 @@ const myFn = new lambda.Function(this, 'Fn', {
...
+
+const deadLetterQueue = new sqs.Queue(this, 'DeadLetterQueue');;
+
+const myFn = new lambda.Function(this, 'Fn', {
+  // other props
+  onFailure: new destinations.SqsDestination(deadLetterQueue)
+})
+```
+
+*Only standard SQS queues are supported for failure destinations, FIFO queues are not supported.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda-destinations/README.md:56]: Can you move this statement to line 20?

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for the docs update @RichardWatkins1.

Couple of comments below.

File: packages/@aws-cdk/aws-lambda-destinations/README.md (hunk 1 above)
Comment (RichardWatkins1) [reply packages/@aws-cdk/aws-lambda-destinations/README.md:20]: Yes sorry you're right 👍

File: packages/@aws-cdk/aws-lambda-destinations/README.md (hunk 2 above)
Comment (RichardWatkins1) [reply packages/@aws-cdk/aws-lambda-destinations/README.md:56]: Will do 👍

**Review by nija-at — CHANGES_REQUESTED** [review]
Hey @RichardWatkins1 - looks like the code sample you've introduced is not compiling. See the build logs in the comment above.

See https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#documentation

**Comment by kaizen3031593** [comment]
@RichardWatkins1, run `yarn rosetta:extract --strict` in `lambda-destinations` to see the errors you are getting. Specifically, you no longer can write `// other props` because we need all required props to be present for it to compile. You also likely have duplicate imports, check `rosetta/default.ts-fixture` for imports that you get for free (your code will be copy/pasted into `/// here`).

File: packages/@aws-cdk/aws-lambda-destinations/README.md [hunk 3]
```diff
@@ -38,6 +38,23 @@ const myFn = new lambda.Function(this, 'Fn', {
...
+
+const deadLetterQueue = new sqs.Queue(this, 'DeadLetterQueue');;
+
+const myFn = new lambda.Function(this, 'Fn', {
+  runtime: lambda.Runtime.NODEJS_12_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromAsset(path.join(__dirname, 'lambda-handler')),
+  // sqs queue for unsuccessful invocations
+  onFailure: new destinations.SqsDestination(deadLetterQueue)
+})
```
Comment (kaizen3031593) [inline packages/@aws-cdk/aws-lambda-destinations/README.md:55]:
```suggestion
import * as sqs from '@aws-cdk/aws-sqs';

const deadLetterQueue = new sqs.Queue(this, 'DeadLetterQueue');

const myFn = new lambda.Function(this, 'Fn', {
  runtime: lambda.Runtime.NODEJS_12_X,
  handler: 'index.handler',
  code: lambda.Code.fromInline('// your code'),
  // sqs queue for unsuccessful invocations
  onFailure: new destinations.SqsDestination(deadLetterQueue),
});
```

**Review by kaizen3031593 — APPROVED** [review]

**Pull request merged**